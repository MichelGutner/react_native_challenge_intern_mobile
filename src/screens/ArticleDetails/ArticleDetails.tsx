import React from 'react';
import { Text, View } from 'react-native';
import HTMLView from 'react-native-htmlview';

const ArticleDetails = props => {
  const article = props.route.params.item;
  console.log('Michel');
  console.log(props.route.params);
  return (
    <View>
      <Text>{'Titulo: ' + article.title}</Text>
      <HTMLView value={article.content} />
    </View>
  );
};

export default ArticleDetails;
