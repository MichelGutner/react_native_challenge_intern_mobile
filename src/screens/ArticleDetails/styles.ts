import { Themes } from './../../../themes/themes';
import styled from 'styled-components/native';

export const Container = styled.ScrollView`
  padding: 20px;
  background-color: ${Themes.colors.whiteOpacity};
`;

export const Title = styled.Text`
  color: ${Themes.colors.backgroundSanar};
  font-size: 26px;
  font-weight: bold;
  margin-top: 10px;
  margin-bottom: 10px;
  align-items: center;
`;

export const Content = styled.View`
  flex: 1;
  background-color: transparent;
  margin: 2px;
`;
