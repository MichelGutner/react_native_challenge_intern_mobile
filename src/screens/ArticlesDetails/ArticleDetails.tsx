import React from 'react';
import { Text, View } from 'react-native';

type Props = {
  route: Parameters;
  params: Parameters;
};

const ArticlesDetails: React.FC<Props> = props => {
  const article = props.route.params.articles;
  return (
    <View>
      <Text>{article.title}</Text>
    </View>
  );
};

export default ArticlesDetails;
