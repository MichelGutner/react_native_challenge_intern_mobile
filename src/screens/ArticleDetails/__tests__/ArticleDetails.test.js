import { NavigationContainer } from '@react-navigation/native';
import React from 'react';
import renderer from 'react-test-renderer';
import ArticleDetails from '../ArticleDetails';

test('renders correctly', () => {
  const route = {
    route: {
      params: { item: { title: 'TITLE', date: '2010/08/2021', lang: 'US' } },
    },
  };
  const tree = renderer
    .create(
      <NavigationContainer>
        <ArticleDetails {...route} />
      </NavigationContainer>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

//  const article = props.route.params.item;
