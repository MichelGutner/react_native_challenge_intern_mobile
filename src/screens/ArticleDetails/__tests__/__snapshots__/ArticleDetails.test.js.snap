// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  style={
    Array [
      Object {
        "backgroundColor": "rgba(255,255,255,0.5)",
        "paddingBottom": 20,
        "paddingLeft": 20,
        "paddingRight": 20,
        "paddingTop": 20,
      },
    ]
  }
>
  <View>
    <Text
      style={
        Array [
          Object {
            "alignItems": "center",
            "color": "#0050F0",
            "fontSize": 26,
            "fontWeight": "bold",
            "marginBottom": 10,
            "marginTop": 10,
          },
        ]
      }
    >
      TITLE
    </Text>
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "marginBottom": 2,
            "marginLeft": 2,
            "marginRight": 2,
            "marginTop": 2,
          },
        ]
      }
    >
      <View>
        <Text
          style={
            Array [
              null,
              null,
            ]
          }
        >
          undefined
        </Text>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
